### ISSUE TITLE
no <link rel=alternates> anymore / seomatic.link.get('alternate') empty after 3.4.0 

### ISSUE THREAD

**Issue by d--j** [nystudio107/craft-seomatic#958]
### Describe the bug

After #535 landed all our entries without individual "Sitemap Enabled" SEO overwrites return an empty array for `seomatic.link.get('alternate')`. (More precisely `DynamicMeta::getLocalizedUrls` is returning an empty array for them.)

### To reproduce
Steps to reproduce the behaviour:
1. A CraftCMS installation with two languages. SEOMatic with `addHrefLang` setting of `true`
2. A Section with a SEO Field with setting `sitemapEnabledFields` that includes `sitemapUrls`. Default value for "Sitemap Enabled" for this section is enabled.
3. Create an entry in this section. Enable it in both languages. Do _not_ enable the overwrite for `sitemapUrls`.
4. The source code of the frontend page does not include any `<link rel="alternate">` or `<meta property="og:locale">` tags. The `seomatic.link.get('alternate')` Twig function returns an empty array.

### Expected behaviour
Since we did not overwrite the "Sitemap Enabled" setting for this entry, the default should be taken and the aforementioned meta/link tags should be enabled.

### Versions
- Plugin version: 3.4.0
- Craft version: 3.7.10

### Hotfix

* Apply f63a316e4db5db6273d30121cfbbc46ab10fee10 (this is probably incomplete – it fixes this particular problem but I suspect the change of #535 will need more places to use `Seomatic::$plugin->helper->isInherited` )
* Downgrade to version `3.3.48` (only possible when you have a DB backup)

**Comment by khalwat** [comment]
Addressed in: https://github.com/nystudio107/craft-seomatic/commit/2a5dd4658380854daa545d405e08096b8e0b72ba

You can try it now by setting your semver in your `composer.json` to look like this:
```json
    "nystudio107/craft-seomatic": "dev-develop as 3.4.3”,
```
Then do a `composer clear-cache && composer update`

I searched the codebase, and didn't find anywhere else that should be affected by this omission.

Thanks for the report!

**Comment by denisyilmaz** [comment]
this problem seems to be still present, or at least its present in GraphQL Endpoints:

```gql
{
  entry(uri:"example-uri", site:"english") {
    id
    siteId
    url
    title
    canonicalId
    seomatic(asArray:true) {
      metaLinkContainer
    }
  }
  languageSwitch: seomatic(uri:"suche", asArray:true, siteId:3) {
      metaLinkContainer
    }
}
```

does not show alternating urls in the `metaLinkContainer` field but these (we have two languages/sites active):

```json
"seomatic": {
        "metaLinkContainer": "{\"canonical\":{\"href\":\"https://example.test:8443/en/example-uri\",\"rel\":\"canonical\"},\"home\":{\"href\":\"https://example.test:8443/en\",\"rel\":\"home\"},\"author\":{\"href\":\"https://example.test:8443/humans.txt\",\"rel\":\"author\",\"type\":\"text/plain\"},\"publisher\":[]}"
      }
```

CraftCMS 3.7.11

SEOmatic		3.4.6
Calendar			3.3.3
Embedded Assets	2.8.0
Entries Subset	1.2.3
Imager X			v3.5.2
Maps			3.9.2
Neo				2.11.9
Redactor			2.8.8
Related Entry Types	v0.3.2
Scout			2.4.2

**Comment by khalwat** [comment]
@denisyilmaz appreciate you finding this issue, but can you post a new issue with the details that are specific to your setup, in terms of SEO Settings field usage, etc.?
